### ISSUE TITLE
Port number 0 does not parse correctly when using an unknown scheme

### ISSUE THREAD

**Issue by ethanshea-ms** [dotnet/runtime#37865]
# Port number 0 does not parse correctly when using an unknown scheme

# General

Reproducing code:

```csharp
Uri.TryCreate("http://example.com:0", UriKind.Absolute, out var uri);	
Console.WriteLine(uri.Port);
Console.WriteLine(uri.IsDefaultPort);
Console.WriteLine(uri);
		
Uri.TryCreate("rtsp://example.com", UriKind.Absolute, out uri);	
Console.WriteLine(uri.Port);
Console.WriteLine(uri.IsDefaultPort);
Console.WriteLine(uri);

Uri.TryCreate("rtsp://example.com:0", UriKind.Absolute, out uri);	
Console.WriteLine(uri.Port);
Console.WriteLine(uri.IsDefaultPort);
Console.WriteLine(uri);
```

In the final example, the port property should be 0, however it is -1.

Note that port 0 is a valid port, and is defined to be reserved by [RFC1340](https://tools.ietf.org/html/rfc1340#page-9)

**Comment by scalablecory** [comment]
@MihaZupan

**Comment by scalablecory** [comment]
This looks like a bug. Are you blocked on this, or was it just something you stumbled on?

**Comment by ethanshea-ms** [comment]
I'm not blocked, I'm using a workaround where I parse the port number myself but this is not ideal.

**Comment by danmosemsft** [comment]
Any interest in investigating a bit?

**Comment by stephentoub** [comment]
(Note this behavior isn't specific to .NET Core; it's the same on .NET Framework.)

**Comment by MihaZupan** [comment]
Looking at the code, I doubt this was intentional. It seems to be an edge-case bug for value 0.

Specifically, this like:
https://github.com/dotnet/runtime/blob/e45d130e9e01c553371ef47f21514ad07f2e0575/src/libraries/System.Private.Uri/src/System/Uri.cs#L2264

Should read
```c#
if (notEmpty && _syntax.DefaultPort != port)
```

As `info.Offset.PortValue` will be 0 (init value) at this point unless the scheme has a defined default port, where as `_syntax.DefaultPort` will hold a constant -1 if no default port is set.

**Comment by karelz** [comment]
@CarnaViire you discussed this with @MihaZupan - it should be very straightforward fix with adding a test. Would be nice to have it in 5.0 given its simplicity.
